packages/fast-components/src/checkbox/checkbox.ts

```diff
@@ -0,0 +1,141 @@
+import { attr, FastElement, observable, Observable } from "@microsoft/fast-element";
+import { FormAssociated } from "../form-associated";
+import { keyCodeSpace } from "@microsoft/fast-web-utilities";
+import { bool } from "../utilities";
+
+/* tslint:disable:member-ordering */
+export class Checkbox extends FormAssociated<HTMLInputElement> {
+    @attr({ attribute: "readonly" })
+    public readOnly: boolean; // Map to proxy element
+    private readOnlyChanged(): void {
+        if (this.proxy instanceof HTMLElement) {
+            this.proxy.readOnly = this.readOnly;
+        }
+    }

```

Comment:
I like how you've set this up, but it obviously requires disabling member ordering. Should we create an issue to disable that all-up in our tslint or in this package? I think this convention is super intuitive and easy to read - I'd like us to adopt it.

Reply:
I agree, I think we could start adding some style conventions to our contributor documentation because we definitely will get contributions that are all over the place.

Reply:
I'll put up a PR.